radix-engine/src/system/system_callback.rs

```diff
@@ -61,6 +64,158 @@ pub type SystemBootSubstate = SystemBoot;
+            VersionedSystemLogic::V1 => {
+                let mut system_service = SystemService::new(api);
+                let intent = executable
+                    .intents()
+                    .get(0)
+                    .expect("This should have been checked in init");
+                let rtn = system_service.call_function(
+                    TRANSACTION_PROCESSOR_PACKAGE,
+                    TRANSACTION_PROCESSOR_BLUEPRINT,
+                    TRANSACTION_PROCESSOR_RUN_IDENT,
+                    scrypto_encode(&TransactionProcessorRunInputEfficientEncodable {
+                        manifest_encoded_instructions: intent.encoded_instructions.clone(),
+                        global_address_reservations,
+                        references: Rc::new(intent.references.clone()),
+                        blobs: intent.blobs.clone(),
+                    })
+                    .unwrap(),
+                )?;
+                let output: Vec<InstructionOutput> = scrypto_decode(&rtn).unwrap();
+                output
+            }
+            VersionedSystemLogic::V2 => {
+                let intents = executable.intents();
+                for intent in intents {
+                    let mut system_service = SystemService::new(api);
+                    let virtual_resources = intent
+                        .auth_zone_init
+                        .simulate_every_proof_under_resources
+                        .clone();
+                    let virtual_non_fungibles =
+                        intent.auth_zone_init.initial_non_fungible_id_proofs.clone();
+                    let auth_zone = AuthModule::create_auth_zone(
+                        &mut system_service,
+                        None,
+                        virtual_resources,
+                        virtual_non_fungibles,
+                    )?;
+
+                    api.kernel_set_call_frame_data(Actor::Function(FunctionActor {
+                        blueprint_id: BlueprintId::new(
+                            &TRANSACTION_PROCESSOR_PACKAGE,
+                            TRANSACTION_PROCESSOR_BLUEPRINT,
+                        ),
+                        ident: TRANSACTION_PROCESSOR_RUN_IDENT.to_string(),
+                        auth_zone,
+                    }))?;
+                }
+
+                {
+                    let mut system_service = SystemService::new(api);
+                    let intent = intents.get(0).unwrap();
+
+                    let txn_processor = TxnProcessor::<InstructionV1>::init(
+                        intent.encoded_instructions.clone(),
+                        global_address_reservations.clone(),
+                        intent.blobs.clone(),
+                        MAX_TOTAL_BLOB_SIZE_PER_INVOCATION,
+                        &mut system_service,
+                    )?;
+                    let output = txn_processor.execute(&mut system_service)?;
+
+                    let owned_nodes = api.kernel_get_owned_nodes()?;
```

Comment:
A lot of this code looks common to setting up / disposing of a call frame - perhaps it should be commonised?